### PR TITLE
Allowing n_params > n_dims in function solve()

### PR DESCRIPTION
**Pull request by piwanek**
For n_params > n_dims we (me and @rpoleski) have tested that this solution works.

### CODE REVIEW

**Comment by JohannesBuchner** [comment]
Thanks. Could you please add a unit test? see https://github.com/JohannesBuchner/PyMultiNest/blob/master/tests/pymultinest_test.py 
The existing tests could be more sophisticated, at the moment they just test that nothing is crashing, although pymultinest_minimal_test.py verifies that the functions are called.

**Comment by rpoleski** [comment]
I think this change also addresses the issue #108 raised by @jluastro.

**Comment by piwanek** [comment]
@JohannesBuchner, requested unit test was added.

File: tests/pymultinest_test.py [hunk 1]
```diff
@@ -38,6 +38,42 @@ def myloglike(cube):
 	assert numpy.isclose(result['logZ'], 236.0, atol=1), result['logZ']
 	assert numpy.isclose(result['logZerr'], 0.1, atol=1), result['logZerr']
 
+def test_2():
```
Comment (JohannesBuchner) [inline tests/pymultinest_test.py]: add here 
	test_2.cube_dim = None
	test_2.params_dim = None

File: tests/pymultinest_test.py [hunk 2]
```diff
@@ -38,6 +38,42 @@ def myloglike(cube):
...
+	"""testing n_params != n_dims case"""
+	def myprior(cube):
+		cube[-1] = numpy.sum(cube[:-1])
+		return cube * 10 * pi
+
+	def myloglike(cube):
```
Comment (JohannesBuchner) [inline tests/pymultinest_test.py]: add here:
	test_2.params_dim = cube.shape

File: tests/pymultinest_test.py [hunk 3]
```diff
@@ -38,6 +38,42 @@ def myloglike(cube):
 	assert numpy.isclose(result['logZ'], 236.0, atol=1), result['logZ']
 	assert numpy.isclose(result['logZerr'], 0.1, atol=1), result['logZerr']
 
+def test_2():
+	"""testing n_params != n_dims case"""
+	def myprior(cube):
```
Comment (JohannesBuchner) [inline tests/pymultinest_test.py]: add here:
	test_2.prior_dim = cube.shape

File: tests/pymultinest_test.py [hunk 4]
```diff
@@ -38,6 +38,42 @@ def myloglike(cube):
...
+		json.dump(parameters, f, indent=2)
+
+	parameters += ['x+y']
+	print()
+	print('evidence: %(logZ).1f +- %(logZerr).1f' % result)
+	print()
+	print('parameter values:')
+	for name, col in zip(parameters, result['samples'].transpose()):
+		print('%15s : %.3f +- %.3f' % (name, col.mean(), col.std()))
+
```
Comment (JohannesBuchner) [inline tests/pymultinest_test.py]: verify here with assert that
test_2.prior_dim == (n_dims,)
test_2.params_dim == (n_params,)

File: tests/pymultinest_test.py [hunk 5]
```diff
@@ -38,6 +38,42 @@ def myloglike(cube):
...
+	assert numpy.isclose(result['logZ'], 239.1, atol=1), result['logZ']
+	assert numpy.isclose(result['logZerr'], 0.1, atol=1), result['logZerr']
+
+
```
Comment (JohannesBuchner) [inline tests/pymultinest_test.py]: and please do the same in test_1

**Comment by piwanek** [comment]
It turns out that prior function gets array of size (n_params,) and likelihood function gets array of size (n_dims,). We corrected chi2 calculation: `cos(cube[:-1] / 2.)` => `cos(cube / 2.)`. Now the evidence test results in the same value. Thanks for these suggestions. First unit test was also modified.

**Comment by rpoleski** [comment]
It seems that all the suggestions by @JohannesBuchner were incorporated. Now, this pull request also improves previously existing unit test.

**Pull request merged**